projects/snapd/yaml_fuzzer.go

```diff
@@ -0,0 +1,11 @@
+package fuzz
+
+import "github.com/snapcore/snapd/snap"
+
+func FuzzInfo(data []byte) int {
+	_, err := snap.InfoFromSnapYaml(data)
```

Comment:
Not sure this function is particularly useful target. With the way it works internally, this will most likely exercise only the gopkg.in/yaml.v2 library. However, if you with to play with it, I think there may be more useful targets that accept some interesting inputs:

- x11 package parses xauth files
- snap-seccomp parses the seccomp templates generated by snapd 
- there's some code for parsing /proc/self/mountinfo
- there's a vendor shell lexer libary under sturil/shlex
- there's some C code in cmd/snap-seccomp that parses os-release and mountinfo
- grubenv parsing code in bootloader/grubenv

I occasionally run go-fuzz on some of the Go package, so far with little success (which is probably a good sign?)

Reply:
Just to mention, snap-seccomp is probably the highest value target as it needs to be setuid root.

Reply:
Awesome. I would be interested targeting other targets as well. Thank you for the suggested targets.

I suggest we start with the initial integration with the current fuzzer and let it run continuously for a few weeks before more targets are added.